libs/execution/include/hpx/execution/executors/thread_pool_executor.hpp

```diff
@@ -90,8 +90,9 @@ namespace hpx { namespace parallel { namespace execution {
             threads::thread_schedule_hint schedulehint, launch policy, F&& f,
             Ts&&... ts)
         {
-            hpx::util::thread_description desc(
-                f, "hpx::parallel::execution::parallel_executor::post");
+            char const* annotation =
+                hpx::traits::get_function_annotation<F>::call(f);
+            hpx::util::thread_description desc(f, annotation);
```

Comment:
@biddisco was this change actually needed? None of the other places doing the same have changed... If I'm reading the constructor for `thread_description` correctly it will first try to use `hpx::traits::get_function_annotation<F>::call(f)`, and only if it's null will it use the fallback (the second argument). If we don't want the fallback that can be left out.